{
  try {
    return stash.get().services();
  }
 catch (  Exception e) {
    throw U.withCause(new InvalidObjectException(e.getMessage()),e);
  }
 finally {
    stash.remove();
  }
}
